unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfileController.cs

```diff
@@ -134,6 +137,6 @@ public UniTask<UserProfile> RequestFullUserProfileAsync(string userId)
         var task = new UniTaskCompletionSource<UserProfile>();
         pendingUserProfileTasks[userId] = task;
 
-        return task.Task;
+        return task.Task.AttachExternalCancellation(cancellationToken);
```

Comment:
Incorrect pattern. Use
```
        cancellationToken.RegisterWithoutCaptureExecutionContext(() => task.TrySetCanceled());
```
instead.

Avoid `AttachExternalCancellation` by all means: it does not cancel, it just stops spinning